contracts/compounder/rewardHandlers/BalancerSwapsHandler.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { SafeERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/IERC20.sol";
+import { IBalancerVault, IAsset } from "../../interfaces/balancer/IBalancerCore.sol";
+import { HandlerBase } from "./HandlerBase.sol";
+
+contract BalancerSwapsHandler is HandlerBase {
+    using SafeERC20 for IERC20;
+
+    // ----------------------------------------------------------------
+    // Types
+    // ----------------------------------------------------------------
+
+    struct SwapPath {
+        bytes32[] poolIds;
+        address[] assetsIn;
+    }
+    SwapPath internal swapPath;
+
+    constructor(
+        address _token,
+        address _strategy,
+        address _balVault,
+        address _wethToken,
+        SwapPath memory _swapPath
+    ) HandlerBase(_token, _strategy, _balVault, _wethToken) {
+        swapPath = _swapPath;
```

Comment:
Shouldn't at least be necessary to check that `poolIds.length + 1 == assetsIn.length` ?

Reply:
Or `==` if you are considering WETH out of the array